prombench/Makefile

```diff
@@ -4,44 +4,58 @@ ifeq ($(AUTH_FILE),)
 AUTH_FILE = /etc/serviceaccount/service-account.json
 endif
 
+ifeq ($(PROVIDER),)
+PROVIDER = gke
+endif
+
 .PHONY: deploy clean
 deploy: nodepool_create resource_apply
 # GCP sometimes takes longer than 30 tries when trying to delete nodepools
 # if k8s resources are not already cleared
 clean: resource_delete nodepool_delete
 
 nodepool_create:
-	$(INFRA_CMD) gke nodepool create -a ${AUTH_FILE} \
-		-v ZONE:${ZONE} -v PROJECT_ID:${PROJECT_ID} -v CLUSTER_NAME:${CLUSTER_NAME} -v PR_NUMBER:${PR_NUMBER} \
-		-f manifests/prombench/nodepools.yaml
+	${INFRA_CMD} ${PROVIDER} nodepool create -a ${AUTH_FILE} \
+		-v ZONE:${ZONE} -v PROJECT_ID:${PROJECT_ID} \
+		-v REGION:${REGION} -v NODE_ROLE:${NODE_ROLE} -v ROLE_ARN:${ROLE_ARN} -v SUBNET_IDS:${SUBNET_IDS} -v SEPARATOR:${SEPARATOR} \
+		-v CLUSTER_NAME:${CLUSTER_NAME} -v PR_NUMBER:${PR_NUMBER} 
+		-f manifests/prombench/nodes_${PROVIDER}.yaml
 
 resource_apply:
-	$(INFRA_CMD) gke resource apply -a ${AUTH_FILE} \
-		-v ZONE:${ZONE} -v PROJECT_ID:${PROJECT_ID} -v CLUSTER_NAME:${CLUSTER_NAME} \
+	$(INFRA_CMD) ${PROVIDER} resource apply -a ${AUTH_FILE} \
+		-v ZONE:${ZONE} -v PROJECT_ID:${PROJECT_ID} \
+		-v REGION:${REGION} \ 
+		-v CLUSTER_NAME:${CLUSTER_NAME} \
 		-v PR_NUMBER:${PR_NUMBER} -v RELEASE:${RELEASE} -v DOMAIN_NAME:${DOMAIN_NAME} \
 		-v GITHUB_ORG:${GITHUB_ORG} -v GITHUB_REPO:${GITHUB_REPO} \
 		-f manifests/prombench/benchmark
 
 # Required because namespace and cluster-role are not part of the created nodepools
 resource_delete:
-	$(INFRA_CMD) gke resource delete -a ${AUTH_FILE} \
-		-v ZONE:${ZONE} -v PROJECT_ID:${PROJECT_ID} -v CLUSTER_NAME:${CLUSTER_NAME} -v PR_NUMBER:${PR_NUMBER} \
+	$(INFRA_CMD) ${PROVIDER} resource delete -a ${AUTH_FILE} \
+		-v ZONE:${ZONE} -v PROJECT_ID:${PROJECT_ID} \
+		-v REGION:${REGION} \
+		-v CLUSTER_NAME:${CLUSTER_NAME} -v PR_NUMBER:${PR_NUMBER} \
 		-f manifests/prombench/benchmark/1c_cluster-role-binding.yaml \
 		-f manifests/prombench/benchmark/1a_namespace.yaml
 
 nodepool_delete:
-	$(INFRA_CMD) gke nodepool delete -a ${AUTH_FILE} \
-		-v ZONE:${ZONE} -v PROJECT_ID:${PROJECT_ID} -v CLUSTER_NAME:${CLUSTER_NAME} -v PR_NUMBER:${PR_NUMBER} \
-		-f manifests/prombench/nodepools.yaml
+	$(INFRA_CMD) ${PROVIDER} nodepool delete -a ${AUTH_FILE} \
+		-v ZONE:${ZONE} -v PROJECT_ID:${PROJECT_ID} \
+		-v REGION:${REGION} -v NODE_ROLE:${NODE_ROLE} -v ROLE_ARN:${ROLE_ARN} -v SUBNET_IDS:${SUBNET_IDS} -v SEPARATOR:${SEPARATOR} \
```

Comment:
`SEPARATOR` would be a good candidate for `DefaultDeploymentVar` :) 

https://github.com/prometheus/test-infra/blob/623686db4f315d316d6ba3bce3c58dc288b14c6c/pkg/provider/provider.go#L45-L51

Then we could just mention that `SEPARATOR` is by default set to `,` and user could change it needed (in the documentation). Then we could remove all other the `SEPARATOR` references from Makefile and README. :)